### DISCUSSION TITLE
Ask a question about SSO

### DISCUSSION THREAD

**Discussion by zhuyifang** [directus/directus#6967] [Archive]
After a user logs in through SSO, the user corresponds to the UID of the SSO Providers. Does directus save it? If not, where should it be and which method is better for storage?

Because, for example, Line supports returning the user’s email, but wechat does not have this option, so I want to understand the logic of this part and save it in directus for later use

**Comment by rijkvanzanten** [comment]
As of right now, Directus relies on the email address coming from the oAuth provider to log the user in. An external ID is not saved. (cc @aidenfoxx)

**Reply by aidenfoxx** [reply]
This would definitely be something we'd do if we created an  `auth-oauth` package as we discussed. I assume the SSO UID would be be the `identifier` field in that scenario.

For a quick fix we could start storing the UID in the `identifier` field as part of directus-core for anyone wishing to consume it.

EDIT: Assuming my PR makes it in. 😛

**Comment by zhuyifang** [comment]
@rijkvanzanten ,@aidenfoxx,Thank you very much for your answers。
For users to log in using SSO, my suggestion is:
We can make the following adjustments in the official login program (/node_modules/directus/dist/controllers/auth.js).
1. If the user id is returned in hooks oauth.${req.params.provider}.login.before, authenticationService.authenticate directly uses the id to generate the related accessToken;
2. If the user id is not returned, follow the existing logic

I'm happy to contribute code, but I don't know how to participate. I'm sorry I use github less

**Reply by rijkvanzanten** [reply]
> If the user id is returned in hooks oauth.${req.params.provider}.login.before, authenticationService.authenticate directly uses the id to generate the related accessToken;

Yes. To add to this, we'll have to store the oAuth providers' unique identifier in a new column under directus users, so we can cross reference the correct ID.
